LibTest/io/Link/create_A01_t01.dart

```diff
@@ -50,6 +53,7 @@ _main(Directory sandbox) async {
   await link.create(target.path).then((Link created) {
     Expect.isTrue(created.existsSync());
     Expect.equals(link.path, created.path);
+    Expect.equals(target.path, created.targetSync());
```

Comment:
Just wondering, is there a danger in mixing sync and async versions of io functions (deadlocks)?

It looks like we have this kind of mixture in some (perhaps lots of) other locations already, but if it was never discussed explicitly then it would be nice to know that it's safe.

Reply:
Tests mainly written usyng sync syntax. But here we are testing async function `create()` so, sync and async code is mixed here. But there were no any failures caused by this. We'll see flaky tests if the code is racy

Reply:
OK, I created https://github.com/dart-lang/sdk/issues/53709 to get some extra input on the sync/async mixture. I think it is safe, but I don't know enough to explain exactly why it would be safe. ;-)

In any case, if we need to do anything then it can be done in another PR.